components/kubearchive/base/monitoring-otel-collector.yaml

```diff
@@ -0,0 +1,102 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: kubearchive-otel-collector
+---
+# Required so kube-rbac-proxy creates tokenreviews and subjectaccessreviews
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: kubearchive-otel-collector-auth-delegator
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:auth-delegator
+subjects:
+  - kind: ServiceAccount
+    name: kubearchive-otel-collector
+    namespace: product-kubearchive
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: otel-collector
+  labels:
+    app: otel-collector
+spec:
+  ports:
+    - name: otlp-http
+      port: 4318
+    - name: metrics
+      port: 8888
+    - name: https
+      port: 8443
+  selector:
+    app: otel-collector
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: otel-collector
+  labels:
+    app: otel-collector
+spec:
+  selector:
+    matchLabels:
+      app: otel-collector
+  template:
+    metadata:
+      labels:
+        app: otel-collector
+    spec:
+      serviceAccount: kubearchive-otel-collector
+      containers:
+        - name: kube-rbac-proxy
+          image: quay.io/brancz/kube-rbac-proxy:v0.19.0
```

Comment:
As far as I understood it, the program decided to move away from using kube-rbac-proxy altogether (rather than find alternative sources for the image).
Recent kubebuilder releases also scaffold services in a way that will not require it. Is it an option here?

Reply:
One of the alternatives listed in the official discussion is to use `quay.io/brancz/kube-rbac-proxy` I'm using (https://github.com/kubernetes-sigs/kubebuilder/discussions/3907). Also I'm following the README.md from the monitoring component on this very same repo, which uses `quay.io/brancz/kube-rbac-proxy` as well.

Reply:
I was not aware the monitoring component readme is referencing this image, but have a look [here](https://docs.google.com/document/d/1B_CBBXvjUXHqTUunB1D7cy7nd6fI0PzRxd3szgEAaSA/edit?tab=t.0) under requirements.

Reply:
I checked on the developers channel and it should be fine, as we can't avoid using the image (because its better than just not using it).